### PR TITLE
delete users

### PR DESCRIPTION
**Pull request by JackScher**

### CODE REVIEW

File: app/templates/users.html [hunk 1]
```diff
@@ -62,7 +62,7 @@
            <svg class="w-4 h-4 mr-2" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M17.414 2.586a2 2 0 00-2.828 0L7 10.172V13h2.828l7.586-7.586a2 2 0 000-2.828z"></path><path fill-rule="evenodd" d="M2 6a2 2 0 012-2h4a1 1 0 010 2H4v10h10v-4a1 1 0 112 0v4a2 2 0 01-2 2H4a2 2 0 01-2-2V6z" clip-rule="evenodd"></path></svg>
             Edit user
           </button>
-          <button type="button" data-modal-toggle="delete-user-modal" class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-red-600 rounded-lg hover:bg-red-800 focus:ring-4 focus:ring-red-300 dark:focus:ring-red-900">
+          <button id={{ user.id }} name="delete-user-btn" type="button" class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-red-600 rounded-lg hover:bg-red-800 focus:ring-4 focus:ring-red-300 dark:focus:ring-red-900">
```
Comment (kipod) [inline app/templates/users.html]: `id` attribute must be unique on HTML page.
use `data-user-id` attribute

File: src/user.ts [hunk 2]
```diff
@@ -45,6 +45,21 @@ export function initUsers() {
 
     const $buttonClose = document.querySelector('#modalCloseButton');
     $buttonClose.addEventListener('click',()=>{modal.hide()})
+
+    const deleteButtons = document.querySelectorAll('button[name="delete-user-btn"]');
+    deleteButtons.forEach((e) => {
+        e.addEventListener("click", async () => {
+          if (confirm("Are sure?")) {
+            const response = await fetch(`/user/delete/${e.id}`, {
+              method: "POST",
```
Comment (kipod) [inline src/user.ts]: POST --> DELETE

File: app/views/user.py [hunk 3]
```diff
@@ -76,3 +76,14 @@ def create():
         flash("User added!", "success")
         user.save()
         return redirect(url_for("user.get_all"))
+
+
+@bp.route("/delete/<id>", methods=["POST"])
+@login_required
+def delete(id):
+    u = m.User.query.filter_by(id=id).first()
+    db.session.delete(u)
```
Comment (kipod) [inline app/views/user.py:90]: add check if `u` is not `None`

File: tests/test_users.py [hunk 4]
```diff
@@ -42,3 +42,13 @@ def test_populate_db(runner: FlaskCliRunner):
...
     assert f"populated by {TEST_COUNT}" in res.stdout
     assert (m.User.query.count() - count_before) == TEST_COUNT
+
+
+def test_delete_user(populate: FlaskClient):
+    login(populate)
+    users = m.User.query.all()
+    uc = len(users)
+    id = users[0].id
+    response = populate.post(f"/user/delete/{id}")
```
Comment (kipod) [inline tests/test_users.py]: post ---> delete

File: tests/test_users.py [hunk 5]
```diff
@@ -42,3 +42,13 @@ def test_populate_db(runner: FlaskCliRunner):
     res: Result = runner.invoke(args=["db-populate", "--count", f"{TEST_COUNT}"])
     assert f"populated by {TEST_COUNT}" in res.stdout
     assert (m.User.query.count() - count_before) == TEST_COUNT
+
+
+def test_delete_user(populate: FlaskClient):
+    login(populate)
+    users = m.User.query.all()
+    uc = len(users)
+    id = users[0].id
```
Comment (kipod) [inline tests/test_users.py]: id = 1

File: tests/test_users.py [hunk 6]
```diff
@@ -42,3 +42,13 @@ def test_populate_db(runner: FlaskCliRunner):
...
+    response = populate.post(f"/user/delete/{id}")
+    assert len(m.User.query.all()) < uc
```
Comment (kipod) [inline tests/test_users.py]: m.User.query.count()

**Review by kipod — CHANGES_REQUESTED** [review]
please fix

**Review by StoliarskyiKostiantyn — APPROVED** [review]

**Review by StoliarskyiKostiantyn — APPROVED** [review]

**Review by kipod — APPROVED** [review]

**Pull request merged**